probes/grpc/grpc.go

```diff
@@ -381,10 +382,24 @@ func (p *Probe) newResult(tgt string) *probeRunResult {
 	}
 }
 
+// setHeaders attaches a list of headers to the given context
+// it iterates over the headers defined in the probe configuration
+func (p *Probe) setHeaders(ctx context.Context) context.Context {
```

Comment:
How about we call it ctxWithHeaders?
```suggestion
func (p *Probe) ctxWithHeaders(ctx context.Context) context.Context {
```

Reply:
sounds good - renamed